contrib/IECoreAppleseed/src/IECoreAppleseed/RendererImplementation.cpp

```diff
@@ -170,16 +170,86 @@ void IECoreAppleseed::RendererImplementation::setOption( const string &name, Con
 
 		if( !valueStr.empty() )
 		{
-			m_project->configurations().get_by_name( "final" )->get_parameters().insert_path( optName.c_str(), valueStr.c_str() );
-			m_project->configurations().get_by_name( "interactive" )->get_parameters().insert_path( optName.c_str(), valueStr.c_str() );
-
-			// if the number of passes is different than one, we need to
-			// switch the shading result framebuffer in the final rendering config.
-			if( optName == "generic_frame_renderer.passes" && !isInteractive() )
+			if( optName == "rendering_threads" )
+			{
+				if( const IntData *numThreadsData = runTimeCast<const IntData>( value.get() ) )
+				{
+					// if numThreads is 0, we want to use all the CPU cores.
+					// We can remove any previous "rendering_threads" param and
+					// let appleseed choose the number of threads to use.
+					if( numThreadsData->readable() == 0 )
+					{
+						m_project->configurations().get_by_name( "final" )->get_parameters().remove_path( optName.c_str() );
+						m_project->configurations().get_by_name( "interactive" )->get_parameters().remove_path( optName.c_str() );
+						return;
+					}
+				}
+				else
+				{
+					msg( Msg::Error, "IECoreAppleseed::RendererImplementation::setOption", "as:cfg:rendering_threads option expects an IntData value." );
+				}
+			}
+			else if( algorithm::ends_with( optName, "max_path_length" ) )
+			{
+				if( const IntData *maxPathLengthData = runTimeCast<const IntData>( value.get() ) )
+				{
+					// if maxPathLength is 0, we want to use and unlimited number of bounces and let
+					// russian roulette terminate paths when their contribution is too low.
+					// We can disable max path lengths by remove any previous param.
+					if( maxPathLengthData->readable() == 0 )
+					{
+						m_project->configurations().get_by_name( "final" )->get_parameters().remove_path( optName.c_str() );
+						m_project->configurations().get_by_name( "interactive" )->get_parameters().remove_path( optName.c_str() );
+						return;
+					}
```

Comment:
Should there be an `else` here where we set the number of threads to the non-zero value?